### PR TITLE
core: disabling cachew through CACHEW_DISABLE env

### PR DESCRIPTION
**Pull request by seanbreckenridge**
went with `fnmatch` since I think that would be nicest to work with in `$PATH` environment variables, though feel free to suggest something else if you think it's better.

https://docs.python.org/3/library/fnmatch.html

### CODE REVIEW

**Comment by seanbreckenridge** [comment]
If you think this is fine, will add some docs

File: src/cachew/__init__.py [hunk 1]
```diff
@@ -486,6 +487,68 @@ def callable_name(func: Callable) -> str:
     mod = getattr(func, '__module__', None) or ''
     return f'{mod}:{func.__qualname__}'
 
+def callable_module_name(func: Callable) -> Optional[str]:
+    return getattr(func, '__module__', None)
+
+# could cache this, but might be worth not to, so the user can change it on the fly?
```
Comment (karlicoss) [inline src/cachew/__init__.py:493]: yeah, probably unnecessary, not that much overhead

File: src/cachew/__init__.py [hunk 2]
```diff
@@ -486,6 +487,68 @@ def callable_name(func: Callable) -> str:
...
+    False
+    >>> _matches_disabled_module('mysomething.else', '')
+    False
+    '''
+    import fnmatch
+
+    if module_name == pattern:
+        return True
+
+    for mp, pp in zip(module_name.split('.'), pattern.split('.')):
```
Comment (karlicoss) [inline src/cachew/__init__.py:535]: hmm, wouldn't that disable the module if the pattern is longer than the module name? 
i.e. `_matches_disabled_module('my.browser', 'my.browser.export')` is `True`?

**Comment by karlicoss** [comment]
hmm `test_concurrent_writes[file]` failed, but it's not cause of your changes -- I retried the CI. Thought I fixed that but apparently not =/

File: src/cachew/__init__.py (hunk 2 above)
Comment (seanbreckenridge) [reply src/cachew/__init__.py]: Ah right, zip just uses the smaller length... will compare the lengths beforehand and if the `len(module_parts) < len(pattern_parts)` will just early exit

**Comment by karlicoss** [comment]
Thanks, looks neat! Up to you if you wanna add docs now, or we could give it a bit more testing first?

**Comment by seanbreckenridge** [comment]
Added it into the `doc/` folder now for, can move it into the `ipynb` file later yeah

**Comment by seanbreckenridge** [comment]
alright should be good... I should really figure out some way to preview the `.md` files locally 😅

**Pull request merged**